libs/agno/agno/knowledge/agent.py

```diff
@@ -44,7 +44,7 @@ def document_lists(self) -> Iterator[List[Document]]:
         raise NotImplementedError
 
     @property
-    def async_document_lists(self) -> AsyncIterator[List[Document]]:
+    async def async_document_lists(self) -> AsyncIterator[List[Document]]:
```

Comment:
So technically a function that returns an async iterator shouldn't itself be async. Our `arun` is actually incorrect

Reply:
But I find it confusing externally. Internally I guess we can stick to the correct way

Reply:
Are the others all correct, just the base is wrong?